src/state/connection-mixin.ts

```diff
@@ -122,18 +122,23 @@ export const connectionMixin = <T extends Constructor<HassBaseEl>>(
             }
             if (notifyOnError) {
               forwardHaptic("failure");
+              const lokalizedErrorMessage = (this as any).hass.localize(
+                `component.${err.translation_domain}.exceptions.${err.translation_key}.message`,
+                err.translation_placeholders
+              );
               const message =
+                lokalizedErrorMessage ||
                 (this as any).hass.localize(
                   "ui.notification_toast.service_call_failed",
                   "service",
                   `${domain}/${service}`
                 ) +
-                ` ${
-                  err.message ||
-                  (err.error?.code === ERR_CONNECTION_LOST
-                    ? "connection lost"
-                    : "unknown error")
-                }`;
+                  ` ${
+                    (lokalizedErrorMessage ? "" : err.message) ||
```

Comment:
```suggestion
                    err.message ||
```
I don't think checking the localized error here is needed because we already check at line `130`